File: pikashop-merchant-interface/src/main/java/com/chao/service/SpecificationService.java
package com.chao.service;
import java.util.List;
import java.util.Map;

import com.chao.common.viewobject.CommonResult;
import com.chao.entity.viewobject.Specification;
import com.chao.mybatis.pojo.SpecificationDo;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SpecificationService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<SpecificationDo> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public CommonResult<SpecificationDo> findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Specification specification);
	
	
	/**
	 * 修改
	 */
	public void update(Specification specification);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Specification findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public CommonResult<SpecificationDo> findPage(SpecificationDo specification, int pageNum, int pageSize);
	
}

File: pikashop-management/src/main/webapp/js/service/specificationService.js
//服务层
app.service('specificationService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../manage/specification/list');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../manage/specification/pages?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../manage/specification/one?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../manage/specification/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../manage/specification/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../manage/specification/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../manage/specification/search?page='+page+"&rows="+rows, searchEntity);
	}    	
	//下拉列表
	this.selectOptionList=function(){
		return $http.get('../manage/specification/selections');
	}
	
});
